kod/object/passive/itematt/weapatt/wapurge.kod

```diff
@@ -71,29 +71,28 @@ messages:
    %%% Effect Functions
       
    ModifyDamage(damage = 0, target = $, wielder = $, lData= $)
-   "Wielder has a 1% chance of blinding target with any hit."
+   "Wielder has a percent chance of purging target with any hit."
    {
       local oSpell;
 
       if random(1,100) < piEffect_percent
       {
-         oSpell = send(SYS,@FindSpellByNum,#num=SID_PURGE);
+         oSpell = Send(SYS,@FindSpellByNum,#num=SID_PURGE);
          
          %% Gotta have something to try to remove....
          if Send(target,@IsEnchanted)
          {
-            % Try to purge off random(25,75)% of the enchantments.
-            if send(oSpell,@DoPurge,#who=target,#iChance=random(25,75))
-            {
-               % Only tell our victim if they lost enchantments.
-               send(target,@MsgSendUser,#message_rsc=purger_worked_target,
-                    #parm1=send(wielder,@GetCapDef),#parm2=send(wielder,@GetName));
-            }
-
-            % Tell the wielder every time we fired so they think it's cool.
-            send(wielder,@MsgSendUser,#message_rsc=purger_worked_user,
-                 #parm1=send(target,@GetCapDef),#parm2=send(target,@GetName));
+            Send(target,@MsgSendUser,#message_rsc=purger_worked_target,
+               #parm1=Send(wielder,@GetCapDef),#parm2=Send(wielder,@GetName));
+
+            % Try to purge off random number of the enchantments
+            % Only tell our victim if they lost enchantments
+            Send(oSpell,@CastSpell,#who=self,#lTargets=[target],#iSpellPower=random(25,75),#bItemCast=TRUE);
```

Comment:
In addition to getting the audio cue, this also triggers a large amount of code in Spell::CastSpell() that wasn't run before.  Just checking that you've looked through it all, and it all looks safe.

Reply:
I feel confident that it's safe. Much of that code will be ignored because the `who` is the weapon attribute and  `bItemCast` is TRUE. The only one that stands out to me is the final call to record the cast of purge against the system statistics. I figured casts from a weapon attribute would be useful data.